src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -608,22 +627,78 @@ else if (assignDepth >= 1) {
      */
     private static boolean unnecessaryParenAroundOperators(DetailAST ast) {
         final int type = ast.getType();
-        final int parentType = ast.getParent().getType();
         final boolean isConditional = TokenUtil.isOfType(type, CONDITIONALS_AND_RELATIONAL);
-        boolean result = TokenUtil.isOfType(parentType, CONDITIONALS_AND_RELATIONAL);
+        final boolean isBitwise = TokenUtil.isOfType(type, BITWISE_BINARY_OPERATORS);
+        final boolean result;
         if (isConditional) {
-            if (type == TokenTypes.LOR) {
-                result = result && !TokenUtil.isOfType(parentType, TokenTypes.LAND);
-            }
-            result = result && !TokenUtil.isOfType(parentType, TokenTypes.EQUAL,
-                TokenTypes.NOT_EQUAL);
+            result = checkConditionalOperator(ast);
+        }
+        else if (isBitwise) {
+            result = checkBitwiseBinaryOperator(ast);
         }
         else {
-            result = result && TokenUtil.isOfType(type, UNARY_AND_POSTFIX);
+            result = TokenUtil.isOfType(type, UNARY_AND_POSTFIX)
+                    && isBitWiseBinaryOrConditionalOperator(ast.getParent().getType());
         }
         return result;
     }
 
+    /**
+     * Check if conditional or logical operator has unnecessary parentheses.
+     *
+     * @param ast to check if surrounded by unnecessary parentheses
+     * @return true if Unnecessary parenthesis
+     */
+    private static boolean checkConditionalOperator(DetailAST ast) {
+        final int type = ast.getType();
+        final int parentType = ast.getParent().getType();
+        final boolean isParentEqualityOperator =
+                TokenUtil.isOfType(parentType, TokenTypes.EQUAL, TokenTypes.NOT_EQUAL);
+        boolean result = true;
```

Comment:
Can we do similar to above, i.e.:
```
final boolean result;
if (type == LOR) {
...
}
else if (type == LAND) {
...
}
else {
...
}
return result;
```
?

Reply:
I think to change `boolean result = true` as `final boolean result;` is not possible 

and i think you suggested to change the name result to hasUnecessaryParenthesis

Reply:
>I think to change boolean result = true as final boolean result; is not possible

Please explain why

>and i think you suggested to change the name result to hasUnecessaryParenthesis

Only do this in the 'unnecessaryParenthesis` method, elsewhere we still do not know if parenthesis are unnecessary yet :)

Reply:
Ahh get it https://github.com/checkstyle/checkstyle/pull/12703#discussion_r1153999664 my mistake

Reply:
done